Sources/Datadog/Core/Upload/DataUploader.swift

```diff
@@ -6,32 +6,52 @@
 
 import Foundation
 
-/// URL of the endpoint for data uploads. It's being validated during its creation.
-internal struct DataUploadURL {
-    let url: URL
+/// Creates URL and adds query items before providing them
+internal class UploadURLProvider {
+    private let rawURL: URL
+    private let dateProvider: DateProvider
 
-    init(endpointURL: String, clientToken: String) throws {
+    private var queryItems: [URLQueryItem] {
+        // batch_time
+        let currentDateInMs = dateProvider.currentDate().timeIntervalSinceReferenceDate
+        let batchTimeQueryItem = URLQueryItem(name: "batch_time", value: "\(currentDateInMs)")
+        // ddsource
+        let ddSourceQueryItem = URLQueryItem(name: "ddsource", value: "mobile")
+
+        return [ddSourceQueryItem, batchTimeQueryItem]
+    }
+
+    var url: URL {
+        var urlComponents = URLComponents(url: rawURL, resolvingAgainstBaseURL: false)
+        urlComponents?.percentEncodedQueryItems = queryItems
+
+        // TODO: RUMM-182 how to handle nil value?
```

Comment:
IMO best we can do is to `guard let` and print appropriate errors on `sdkLogger` and `userLogger`.